lebedigital/mapping/emodul_mapping.py

```diff
@@ -1,254 +1,199 @@
+
+    '''
+
+    # load metadata and get the keys
+    metadata = load_metadata(metadataPath)
+    keys = list(metadata.keys())  
+
+    # generate ID for the e-module metadata
+    specimenID = str(uuid.uuid4())
+
+    # read in the ontology as text linewise, creating a list of lines
+    with open(ontoPath, 'r') as file:
+        lines = file.readlines()
+
+        # Set up logger
+        logger.debug('S T A R T')
+        logger.debug('Loaded ttl-File has ' + str(len(lines)) + ' lines.')
+        usedKeys = [] # to count keys that found a placeholder
+        ontoPHcounter = [] # to count all placeholders
+        remainingPH = [] # to count the placeholders that recieved no data
+
+        # iterating through the list of lines
+        for i in range(len(lines)):
+
+            # create a list of placeholders
+            if '_Value$$' in lines[i]:
+                ph = lines[i].split("$$")[1]
+                ontoPHcounter.append(ph)
+
+            # iterate through list of metadata-keys
+            for key in keys:
+
+                placeholder = generate_placeholder(key)
+
+                # if placeholder is in line, replace it with metadata
+                if placeholder in lines[i]:
+                    logger.debug('Found placeholder "' + placeholder + '" for key "' \
+                                 + key + '" with value "' + str(metadata[key]) + '".')
+                    lines[i] = lines[i].replace(placeholder, str(metadata[key]))
+                    usedKeys.append(key)
+
+                # append the specimen-ID name to "key"_ , works for most keys, except 
+                # some keys below
+                key_ = key + "_ "
+                if key_ in lines[i]:
+                    lines[i] = lines[i].replace(key_, key + "_" + str(specimenID) + " ")
+
+
+            # append the specimen-ID name to the exceptions 
+            if "_," in lines[i]:
+                #logger.debug('Appended specimen-ID in line ' + str(i + 1) \
+                #             + ' to ' + str(lines[i].split("_,")[0] + "_,") + '".')    
+                lines[i] = lines[i].replace("_,", "_" + str(specimenID) + ",")
+            if "_ " in lines[i]:
+                #logger.debug('Appended specimen-ID in line ' + str(i + 1) \
+                #             + ' to ' + str(lines[i].split("_ ")[0] + "_ ") + '".')    
+                lines[i] = lines[i].replace("_ ", "_" + str(specimenID) + " ")
+
+
+            # ID-key is not given by metadata but created in this script, so map it now:
+            if generate_placeholder("SpecimenID") in lines[i]:
+                    logger.debug('Found placeholder "' + generate_placeholder("SpecimenID")+ '".')
+                    lines[i] = lines[i].replace(generate_placeholder("SpecimenID"), str(specimenID))
+
+
+            # depending on shape of specimen, set null pointers            
+            if "SpecimenHeight" not in keys:
```

Comment:
This is all very specific for the Youngs modulus test. We can merge that to have something working, but as soon as we move to the next ones (mix), it would be nice to make it general. This means replace all placeholders, remove those lines with empty placeholders, and add uuid automatically (e.g. by adding a specific placeholder). Could you make an issue on that?

Reply:
I mean the mapping that you currently write just replaces in a given ttl file the corresponding strings from another dict also given as a parameter, so the input is a ttl file and a dict, the output is another ttl file (with the replacements made). IMO, this is not specific to a Youngsmodulus tests and as such could be reused for any mapping (mixture, ...).

Reply:
What are the things that have to be specific? Can we just move all that to the extraction script? This would really improve the simplicity of the approach and in particular the coding requirements (because the only thing is the drawio file (and the corresponding ttl) and the json metadata structure, no python code would have to be touched.